tests/context_cli.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+"""
+Test CLI for hierarchical commands with Click context passing.
+
+This CLI has a parent command that sets ctx.obj and child commands
+that access that context, demonstrating hierarchical tool execution.
+"""
+
+import click
+from click_mcp import click_mcp
+
+
+@click_mcp(server_name="context-test-cli")
```

Comment:
Extend this test app with multiple nested levels. Add parameters and context to every level. Then update the tests to verify deeply nested commands and context work.